### PR TITLE
Bump AA, Nemo, and Hecke compat

### PR DESCRIPTION
**Pull request by lgoettgens**
Still waiting for a Hecke release.

### CODE REVIEW

**Comment by lgoettgens** [comment]
This is now waiting for a GAP.jl release with https://github.com/oscar-system/GAP.jl/pull/1042.
cc @fingolfin

File: src/Rings/AbelianClosure.jl [hunk 1]
```diff
@@ -625,25 +625,39 @@ end
...
+
+function add!(a::QQAbFieldElem, b::QQAbFieldElem)
+  a, b = make_compatible(a, b)
+  a.data = add!(a.data, b.data)
+  return a
 end
 
 function neg!(a::QQAbFieldElem)
-  mul!(a.data,a.data,-1)
+  a.data = mul!(a.data, -1)
```
Comment (fingolfin) [inline src/Rings/AbelianClosure.jl]:
```suggestion
  a.data = neg!(a.data)
```

**Review by fingolfin — APPROVED** [review]

**Comment by fingolfin** [comment]
The changes in straight line programs lead to test failures. Haven't yet tried to understand why, need sleep

File: experimental/GModule/test/runtests.jl [hunk 2]
```diff
@@ -220,8 +220,8 @@ end
 
   l2 = prime_decomposition(maximal_order(k), 2)
   k2, _ = Hecke.completion(k, l2[1][1], 120)
-  z = Hecke.local_fundamental_class_serre(k2, prime_field(k2))
-  C, mG, mU = Oscar.GrpCoh.gmodule(k2, prime_field(k2))
+  z = Hecke.local_fundamental_class_serre(k2, Hecke.absolute_base_field(k2))
+  C, mG, mU = Oscar.GrpCoh.gmodule(k2, Hecke.absolute_base_field(k2))
```
Comment (joschmitt) [inline experimental/GModule/test/runtests.jl]: I think "we" "forgot" to export `absolute_base_field`...

File: Project.toml [hunk 3]
```diff
@@ -25,16 +25,16 @@ UUIDs = "cf7118a7-6976-5b1a-9a39-7adc72f591a4"
 cohomCalg_jll = "5558cf25-a90e-53b0-b813-cadaa3ae7ade"
 
 [compat]
-AbstractAlgebra = "0.42.5"
+AbstractAlgebra = "0.43.1"
 AlgebraicSolving = "0.7.0"
 Distributed = "1.6"
 GAP = "0.11.3"
-Hecke = "0.33.0"
+Hecke = "0.34.0"
```
Comment (lgoettgens) [inline Project.toml]:
```suggestion
Hecke = "0.34.1"
```

File: experimental/GModule/src/GaloisCohomology.jl [hunk 4]
```diff
@@ -340,7 +340,7 @@ defaulting to the full automorphism group over the prime field.
 Find the embedding of Gp -> G, realizing the local automorphism group
 as a subgroup of the global one.
 """
-function Oscar.decomposition_group(K::AbsSimpleNumField, mK::Map, mG::Map = automorphism_group(K)[2], mGp::Map = automorphism_group(codomain(mK), prime_field(codomain(mK))); _sub::Bool = false)
+function Oscar.decomposition_group(K::AbsSimpleNumField, mK::Map, mG::Map = automorphism_group(K)[2], mGp::Map = automorphism_group(codomain(mK), Hecke.absolute_base_field(codomain(mK))); _sub::Bool = false)
```
Comment (lgoettgens) [inline experimental/GModule/src/GaloisCohomology.jl]:
```suggestion
function Oscar.decomposition_group(K::AbsSimpleNumField, mK::Map, mG::Map = automorphism_group(K)[2], mGp::Map = automorphism_group(codomain(mK), absolute_base_field(codomain(mK))); _sub::Bool = false)
```

File: experimental/GModule/src/GaloisCohomology.jl [hunk 5]
```diff
@@ -998,7 +998,7 @@ function idele_class_gmodule(k::AbsSimpleNumField, s::Vector{Int} = Int[]; redo:
   Hecke.popindent()
   @vprint :GaloisCohomology 2 " .. gathering the local modules ..\n"
   Hecke.pushindent()
-  C = [gmodule(x[1], prime_field(x[1])) for x = L];
+  C = [gmodule(x[1], Hecke.absolute_base_field(x[1])) for x = L];
```
Comment (lgoettgens) [inline experimental/GModule/src/GaloisCohomology.jl]:
```suggestion
  C = [gmodule(x[1], absolute_base_field(x[1])) for x = L];
```

File: experimental/GModule/src/GaloisCohomology.jl [hunk 6]
```diff
@@ -1314,7 +1314,7 @@ function local_index(CC::Vector{GrpCoh.CoChain{2, PermGroupElem, GrpCoh.MultGrpE
     cn = data[7]
   else
     L, mL = completion(k, P)#, 40*ramification_index(P))
-    C, mGp, mU = gmodule(L, prime_field(L))
+    C, mGp, mU = gmodule(L, Hecke.absolute_base_field(L))
```
Comment (lgoettgens) [inline experimental/GModule/src/GaloisCohomology.jl]:
```suggestion
    C, mGp, mU = gmodule(L, absolute_base_field(L))
```

File: experimental/GModule/src/GaloisCohomology.jl [hunk 7]
```diff
@@ -1860,7 +1860,7 @@ function serre(A::IdeleParent, P::AbsNumFieldOrderIdeal)
   Kp, mKp, mGp, mUp, pro, inj = completion(A, P)
   mp = decomposition_group(A.k, mKp, A.mG, mGp)
   qr = restrict(C, mp)
-  s = Hecke.Hecke.local_fundamental_class_serre(Kp, prime_field(Kp))
+  s = Hecke.Hecke.local_fundamental_class_serre(Kp, Hecke.absolute_base_field(Kp))
```
Comment (lgoettgens) [inline experimental/GModule/src/GaloisCohomology.jl]:
```suggestion
  s = Hecke.Hecke.local_fundamental_class_serre(Kp, absolute_base_field(Kp))
```

File: experimental/GModule/test/runtests.jl (hunk 2 above)
Comment (lgoettgens) [inline experimental/GModule/test/runtests.jl]:
```suggestion
  z = Hecke.local_fundamental_class_serre(k2, absolute_base_field(k2))
  C, mG, mU = Oscar.GrpCoh.gmodule(k2, absolute_base_field(k2))
```

**Pull request merged**